### ISSUE TITLE
Error when limits are set via a function

### ISSUE THREAD

**Issue by trekonom** [eclarke/ggbeeswarm#57]
`geom_beeswarm` breaks when setting the `limits` for a (continuous) positional scale via a function:

My use case is to extend the limits of the y-axis via ` scales::breaks_extended`

``` r
library(ggbeeswarm)
#> Loading required package: ggplot2

ggplot(mtcars, aes(x = factor(1), y = mpg, color = factor(cyl))) +
  geom_beeswarm() +
  scale_y_continuous(limits = function(x) range(scales::breaks_extended(only.loose = TRUE)(x)))
#> Error in x[0L]: object of type 'closure' is not subsettable
```

The same issue arises in case of `scale_x_continuous`:

``` r
library(ggbeeswarm)
#> Loading required package: ggplot2

ggplot(mtcars, aes(x = 1, y = mpg, color = factor(cyl))) +
  geom_beeswarm() +
  scale_x_continuous(limits = function(x) range(scales::breaks_extended(only.loose = TRUE)(x)))
#> Error in x[0L]: object of type 'closure' is not subsettable
```

I also checked for a different function where I use `floor` and `ceiling` to "round" the limits:

``` r
library(ggbeeswarm)
#> Loading required package: ggplot2

ggplot(mtcars, aes(x = factor(1), y = mpg, color = factor(cyl))) +
  geom_beeswarm() +
  scale_y_continuous(limits = function(x) c(floor(x[1]), ceiling(x[2])))
#> Error in x[0L]: object of type 'closure' is not subsettable
```

**Comment by eclarke** [comment]
Thanks for the bug report and apologies for the delay. I'll look into this.

**Comment by eclarke** [comment]
Fixed in #60
